blockmanager.go

```diff
@@ -764,8 +777,7 @@ func (b *blockManager) fetchHeaderBlocks() {
 				"fetch: %v", err)
 		}
 		if !haveInv {
-			b.requestedBlocks[*node.sha] = struct{}{}
-			b.syncPeer.requestedBlocks[*node.sha] = struct{}{}
+			b.requestedBlocks[*node.sha] = b.syncPeer
```

Comment:
Yeah, will refactor the structs to categorize and update all places where they're used.